DatadogInternal/Sources/Utils/DDError.swift

```diff
@@ -90,3 +90,48 @@ public struct InternalError: Error, CustomStringConvertible {
         self.description = description
     }
 }
+
+public struct ObjcException: Error {
+    /// A closure to catch Objective-C runtime exception and rethrow as `Swift.Error`.
+    ///
+    /// - Important: Does nothing by default, it must be set to an Objective-C interopable function.
+    ///
+    /// - Warning: As stated in [Objective-C Automatic Reference Counting (ARC)](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#exceptions),
+    /// in Objective-C, ARC is not exception-safe and  does not perform releases which would occur at the end of a
+    /// full-expression if that full-expression throws an exception. Therefore, ARC-generated code leaks by default
+    /// on exceptions.
+    public static var rethrow: ((() -> Void) throws -> Void) = { $0() }
+
+    /// The underlying `NSError` describing the `NSException`
+    /// thrown by Objective-C runtime.
+    public let error: Error
+}
+
+/// Rethrow Objective-C runtime exception as `Swift.Error`.
+///
+/// - Warning: As stated in [Objective-C Automatic Reference Counting (ARC)](https://clang.llvm.org/docs/AutomaticReferenceCounting.html#exceptions),
+/// in Objective-C, ARC is not exception-safe and  does not perform releases which would occur at the end of a
+/// full-expression if that full-expression throws an exception. Therefore, ARC-generated code leaks by default
+/// on exceptions.
+/// - throws: `ObjcException` if an exception was raised by the Objective-C runtime.
+@discardableResult
+public func objc_rethrow<T>(_ block: () throws -> T) throws -> T {
```

Comment:
change-request/ It seems we're missing explicit tests coverage for the logic in this code. What we test in `ObjcExceptionHandlerTests.swift` is the direct obj-c implementation, bypassing this function. Wouldn't it make sense to change it to utilize `objc_rethrow` instead?